tests/test_dithering_binning.py

```diff
@@ -0,0 +1,169 @@
+        """Test db in values in reversed order"""
+        self.x.reverse()
+        self.db_object.setup_coins(self.x, self.weights)
+        self.db_object.setup_bins(self.labels, self.label_length, len(self.x))
+        self.db_object.distribution_by_value()
+
+        self.assertEqual(0, self.db_object.bins[0].coins[8].value)
+        self.assertEqual(1, self.db_object.bins[0].coins[7].value)
+        self.assertEqual(2, self.db_object.bins[0].coins[6].value)
+        self.assertEqual(3, self.db_object.bins[1].coins[5].value)
+        self.assertEqual(4, self.db_object.bins[1].coins[4].value)
+        self.assertEqual(5, self.db_object.bins[1].coins[3].value)
+        self.assertEqual(6, self.db_object.bins[2].coins[2].value)
+        self.assertEqual(7, self.db_object.bins[2].coins[1].value)
+        self.assertEqual(8, self.db_object.bins[2].coins[0].value)
+
+    def test_negative_distribution_by_value(self):
+        """DB with negative value added."""
+        self.x.append(-10)
+        self.weights.append(1)
+        self.db_object.setup_coins(self.x, self.weights)
+        self.db_object.setup_bins(self.labels, self.label_length, len(self.x))
+        self.db_object.distribution_by_value()
+
+    def test_dithering_balanced_values(self):
+        """Normal values to test"""
+        self.db_object.binning(self.x, self.weights, self.labels, self.label_length)
+        self.assertEqual(0, self.db_object.bins[0].coins[0].value)
+        self.assertEqual(1, self.db_object.bins[0].coins[1].value)
+        self.assertEqual(2, self.db_object.bins[0].coins[2].value)
+        self.assertEqual(3, self.db_object.bins[1].coins[3].value)
+        self.assertEqual(4, self.db_object.bins[1].coins[4].value)
+        self.assertEqual(5, self.db_object.bins[1].coins[5].value)
+        self.assertEqual(6, self.db_object.bins[2].coins[6].value)
+        self.assertEqual(7, self.db_object.bins[2].coins[7].value)
+        self.assertEqual(8, self.db_object.bins[2].coins[8].value)
+
+    def test_labeling(self):
+        """Check if Labels came out right"""
+        self.x.append(float('nan'))  # Handling None and Nan Values
+        self.x.append(None)
+        self.weights.append(1)
+        self.weights.append(1)
+        self.db_object.binning(self.x, self.weights, self.labels, self.label_length)
+
+        self.assertEqual('b1', self.db_object.label[0])
+        self.assertEqual('b1', self.db_object.label[1])
+        self.assertEqual('b1', self.db_object.label[2])
+        self.assertEqual('b1', self.db_object.label[3])
+        self.assertEqual('b1', self.db_object.label[4])
+        self.assertEqual('b2', self.db_object.label[5])
+        self.assertEqual('b2', self.db_object.label[6])
+        self.assertEqual('b3', self.db_object.label[7])
+        self.assertEqual('b3', self.db_object.label[8])
+        self.assertEqual('NaN', self.db_object.label[9])
+        self.assertEqual('NaN', self.db_object.label[10])
```

Comment:
looking pretty dry here, you should put your expected results into a list and use a loop with a single assertion